src/app/modules/analytics-config/components/edit-histogram-label/edit-histogram-label.component.ts

```diff
@@ -0,0 +1,229 @@
+    dataFieldValue: AbstractControl;
+    dataFieldFunction?: AbstractControl;
+    labelControl: AbstractControl;
+    unitControl: AbstractControl;
+    message?: BehaviorSubject<string>;
+    readonly?: BehaviorSubject<boolean>;
+}
+
+@Component({
+  selector: 'arlas-edit-histogram-label',
+  templateUrl: './edit-histogram-label.component.html',
+  styleUrls: ['./edit-histogram-label.component.scss']
+})
+export class EditHistogramLabelComponent implements OnInit, OnDestroy {
+    /**
+     * field from parent form control
+     */
+    @Input() public dataStep: { aggregation: FormGroup; metric: FormGroup; collection: FormControl; };
+    /**
+     * field from parent form control
+     */
+    @Input() public unmanagedFieldRenderStep: {
+        chartXLabel: FormControl;
+        chartYLabel: FormControl;
+        xUnit: FormControl;
+        yUnit: FormControl;
+    };
+    @Input() public data: LabelConfig[];
+    public displayedColumns = ['recap', 'fieldLabel', 'fieldColumn', 'fieldUnits'];
+    /**
+     *  X axis config
+     * @protected
+     */
+    protected xAxisConfig: LabelConfig;
+    /**
+     *  Y axis config
+     * @protected
+     */
+    protected yAxisConfig: LabelConfig;
+    /**
+     * Get lookAndField value to listen collection unit change
+     * @protected
+     */
+    protected lookAndFeelFormControl: AbstractControl<any>;
+    /**
+     * Comparing value
+     * @protected
+     */
+    protected currentCollectionName: string;
+
+    private _destroyed$ = new Subject();
+
+    public constructor(private mainFromService: MainFormService) {
+    }
+
+    public ngOnInit(): void {
+      if (!this.dataStep && !this.unmanagedFieldRenderStep) {
+        console.error('no config found');
+        return;
+      }
+
+      this.currentCollectionName = this.dataStep.collection.value;
+
+      this.xAxisConfig = {
+        title: marker('x-label'),
+        dataFieldValue: this.dataStep.aggregation.get('aggregationField'),
+        labelControl: this.unmanagedFieldRenderStep.chartXLabel,
+        unitControl: this.unmanagedFieldRenderStep.xUnit,
+        message: new BehaviorSubject(''),
+        readonly: new BehaviorSubject(false)
+      };
+
+      this.yAxisConfig = {
+        title: marker('y-label'),
+        dataFieldValue: this.dataStep.metric.get('metricCollectField'),
+        dataFieldFunction: this.dataStep.metric.get('metricCollectFunction'),
+        labelControl: this.unmanagedFieldRenderStep.chartYLabel,
+        unitControl: this.unmanagedFieldRenderStep.yUnit,
+        message: new BehaviorSubject(''),
+        readonly: new BehaviorSubject(false)
+      };
+
+      if (this.xFieldIsDate()) {
+        this.disableXUnitField();
+      }
+
+      if (this.metricCollectionIsCount()) {
+        this.disableYUnitField();
+        this.setYUnitFieldWithCollectionUnit();
+      }
+
+      this.data = [this.xAxisConfig, this.yAxisConfig];
+      this.listenChanges();
+    }
+
+    private listenChanges() {
```

Comment:
I think a little comment to make what the method does more explicit would be great !

Reply:
I agree !